exporter/trace/stackdriver/stackdriver.go

```diff
@@ -0,0 +1,258 @@
+	// for now. The minimum number of workers is 1.
+	NumberOfWorkers int
+}
+
+// WithProjectID sets Google Cloud Platform project as projectID.
+// Without using this option, it automatically detects the project ID
+// from the default credential detection process.
+// Please find the detailed order of the default credentail detection proecess on the doc:
+// https://godoc.org/golang.org/x/oauth2/google#FindDefaultCredentials
+func WithProjectID(projectID string) func(o *options) {
+	return func(o *options) {
+		o.ProjectID = projectID
+	}
+}
+
+// WithOnError sets the hook to be called when there is an error
+// occurred on uploading the span data to Stackdriver.
+// If no custom hook is set, errors are logged.
+func WithOnError(onError func(err error)) func(o *options) {
+	return func(o *options) {
+		o.OnError = onError
+	}
+}
+
+// WithTraceClientOptions sets additionial client options for tracing.
+func WithTraceClientOptions(opts []option.ClientOption) func(o *options) {
+	return func(o *options) {
+		o.TraceClientOptions = opts
+	}
+}
+
+// WithContext sets the context that trace exporter and metric exporter
+// relies on.
+func WithContext(ctx context.Context) func(o *options) {
+	return func(o *options) {
+		o.Context = ctx
+	}
+}
+
+func (o *options) handleError(err error) {
+	if o.OnError != nil {
+		o.OnError(err)
+		return
+	}
+	log.Printf("Failed to export to Stackdriver: %v", err)
+}
+
+const defaultTimeout = 5 * time.Second
+
+// Exporter is a trace exporter that uploads data to Stackdriver.
+//
+// TODO(yoshifumi): add a metrics exporter once the spec definition
+// process and the sampler implementation are done.
+type Exporter struct {
+	traceExporter *traceExporter
+}
+
+// NewExporter creates a new Exporter thats implements trace.Exporter.
+//
+// TODO(yoshifumi): add a metrics exporter one the spec definition
+// process and the sampler implementation are done.
+func NewExporter(opts ...Option) (*Exporter, error) {
+	o := options{}
+	for _, opt := range opts {
+		opt(&o)
+	}
+	if o.ProjectID == "" {
+		ctx := o.Context
+		if ctx == nil {
+			ctx = context.Background()
+		}
+		creds, err := google.FindDefaultCredentials(ctx, traceapi.DefaultAuthScopes()...)
+		if err != nil {
+			return nil, fmt.Errorf("Stackdriver: %v", err)
+		}
+		if creds.ProjectID == "" {
+			return nil, errors.New("Stackdriver: no project found with application default credentials")
+		}
+		o.ProjectID = creds.ProjectID
+	}
+	te, err := newTraceExporter(&o)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Exporter{
+		traceExporter: te,
+	}, nil
+}
+
+func newContextWithTimeout(ctx context.Context, timeout time.Duration) (context.Context, func()) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	if timeout <= 0 {
+		timeout = defaultTimeout
+	}
+	return context.WithTimeout(ctx, timeout)
+}
+
+// Register registers e to the tracer.
+func (e *Exporter) Register() {
+	trace.Register()
+	trace.RegisterExporter(e)
+}
```

Comment:
trace.Register() should be called by application.
RegisterExporter will no longer be there after #162 is merged.
Exporter should be used with either SimpleSpanProcessor or BatchSpanProcessor. BatchSpanProcessor would be more appropriate.

Reply:
Done